docs/configuration_options.rst

```diff
@@ -1,100 +1,110 @@
+  The number of records to process as a batch when writing to Elasticsearch.
 
   * Type: int
   * Default: 2000
   * Importance: medium
 
-``max.in.flight.requests``
-  The maximum number of incomplete batches each task will send before blocking.
+``max.buffered.records``
+  The maximum number of records each task will buffer before blocking acceptance of more records. This config can be used to limit the memory usage for each task.
 
   * Type: int
-  * Default: 5
-  * Importance: medium
+  * Default: 20000
+  * Importance: low
 
-``flush.timeout.ms``
-  The timeout when flushing data to Elasticsearch.
+``linger.ms``
+  Linger time in milliseconds for batching.
+
+  Records that arrive in between request transmissions are batched into a single bulk indexing request, based on the ``batch.size`` configuration. Normally this only occurs under load when records arrive faster than they can be sent out. However it may be desirable to reduce the number of requests even under light load and benefit from bulk indexing. This setting helps accomplishes that - rather than immediately sending out a record, the task will wait upto the given delay to allow other records to be added so that they can be batched into a single request.
 
   * Type: long
-  * Default: 10000
+  * Default: 1
   * Importance: low
 
-``linger.ms``
-  The task groups together any records that arrive in between request transmissions into a single batched request. Normally this occurs only under load when records arrive faster than they can be sent out. However in some circumstances the tasks may want to reduce the number of requests even under moderate load. This setting accomplishes this by adding a small amount of artificial delay. Rather than immediately sending out a record the task will wait for up to the given delay to allow other records to be sent so that the sends can be batched together.
+``flush.timeout.ms``
+  The timeout in milliseconds when flushing records to Elasticsearch.
 
   * Type: long
-  * Default: 1
+  * Default: 10000
   * Importance: low
 
-``max.buffered.records``
-  Approximately the max number of records each task will buffer. This config controls the memory usage for each task.
+``max.in.flight.requests``
+  The maximum number of indexing requests that can be in-flight to Elasticsearch before blocking further requests.
 
   * Type: int
-  * Default: 20000
-  * Importance: low
+  * Default: 5
+  * Importance: medium
 
 ``max.retries``
-  The max allowed number of retries.
+  The maximum number of retries that are allowed.
 
   * Type: int
   * Default: 5
   * Importance: low
 
 ``retry.backoff.ms``
-  The amount of time to wait before attempting to retry a failed batch. This avoids repeatedly sending requests in a tight loop under some failure scenarios.
+  How long to wait in milliseconds, before attempting to retry a failed batch. This avoids retrying requests in a tight loop under certain failure scenarios.
```

Comment:
The way this is phrased makes it sound like there are some failure scenarios where we _will_ retry requests in a tight loop. Is that intentional?